app/utils.py

```diff
@@ -74,25 +75,36 @@ def upload_code(args):
         print("inference_file key not found in config.yaml!")
         return
 
-    # make a zip file of all the files in the root directory and save it in the current directory
-    shutil.make_archive(existing_data['project_name'], 'zip', os.getcwd())
-
-    # upload the code to the server by making a POST request to the server with the zip file and the secret token
-    files = {'file': open(existing_data['project_name'] + '.zip', 'rb')}
     token = keyring.get_password(
         existing_data['project_name'], "project_token")
 
     if token is None:
         print("project token not found!")
         return
 
+    # make a zip file of all the files in the root directory and save it in the current directory
+    shutil.make_archive(token, 'zip', os.getcwd())
+
+    # Generate a sha256 hash of the zip file
+    sha256_hash = hashlib.sha256()
+    with open(token + '.zip', "rb") as f:
+        # Read and update hash string value in blocks of 4K
```

Comment:
The comment literally explains what's going on. `Read and update hash string value in blocks of 4K token` because our files could be huge so we don't generate hash directly we do it in blocks <3